archive/ar_wrap.c

```diff
@@ -400,5 +414,10 @@ int read_write_file(dbfile_info *f, void *writer_ctx, writer_cb writer)
 
     f->offset = lseek(fd, 0, SEEK_CUR);
 
-    return 0;
+done:
+    if (fd) {
```

Comment:
A zero fd is valid. Though in practice the code is absolutely fine, `if (fd != -1)` or `if (fd >= 0)` is more idiomatic.

Reply:
Thanks for catching this! Updated to use `if (fd != -1)`.